geonode.layers.api.tests
========================

.. py:module:: geonode.layers.api.tests


Attributes
----------

.. autoapisummary::

   geonode.layers.api.tests.logger


Classes
-------

.. autoapisummary::

   geonode.layers.api.tests.DatasetsApiTests


Module Contents
---------------

.. py:data:: logger

.. py:class:: DatasetsApiTests

   Bases: :py:obj:`rest_framework.test.APITestCase`


   .. py:attribute:: fixtures
      :value: ['initial_data.json', 'group_test_data.json', 'default_oauth_apps.json']



   .. py:method:: setUp()


   .. py:method:: test_datasets()

      Ensure we can access the Layers list.



   .. py:method:: test_extra_metadata_included_with_param()


   .. py:method:: test_get_dataset_related_maps_and_maplayers()


   .. py:method:: test_raw_HTML_stripped_properties()

      Ensure "raw_*" properties returns no HTML or carriage-return tag



   .. py:method:: test_layer_replace_anonymous_should_raise_error()


   .. py:method:: test_layer_replace_should_redirect_for_not_accepted_method()


   .. py:method:: test_layer_replace_should_raise_error_if_layer_does_not_exists()


   .. py:method:: test_dataset_append_anonymous_should_raise_error()


   .. py:method:: test_dataset_append_should_redirect_for_not_accepted_method()


   .. py:method:: test_dataset_append_should_raise_error_if_layer_does_not_exists()


   .. py:method:: test_layer_replace_should_work(_validate_input_source)


   .. py:method:: test_patch_point_of_contact()


   .. py:method:: test_patch_metadata_author()


   .. py:method:: test_patch_processor()


   .. py:method:: test_patch_publisher()


   .. py:method:: test_patch_custodian()


   .. py:method:: test_patch_distributor()


   .. py:method:: test_patch_resource_user()


   .. py:method:: test_patch_resource_provider()


   .. py:method:: test_patch_originator()


   .. py:method:: test_patch_principal_investigator()


   .. py:method:: test_metadata_update_for_not_supported_method()


   .. py:method:: test_metadata_update_for_not_authorized_user()


   .. py:method:: test_unsupported_file_throws_error()


   .. py:method:: test_valid_metadata_file_with_different_uuid()


   .. py:method:: test_permissions_for_not_permitted_user()


   .. py:method:: test_permissions_for_permitted_user()


   .. py:method:: test_valid_metadata_file()


   .. py:method:: test_download_api()


